cumulusci/tasks/salesforce/update_dependencies.py

```diff
@@ -93,268 +79,165 @@ def _init_options(self, kwargs):
-                ref=dependency.get("ref"),
-            )
-
-        if zip_src:
-            # determine whether to inject namespace prefixes or not
-            options = dependency.copy()
-            if "unmanaged" not in options:
-                namespace = options.get("namespace_inject")
-                options["unmanaged"] = (
-                    not namespace
-                ) or namespace not in self.org_config.installed_packages
-
-            package_zip = MetadataPackageZipBuilder.from_zipfile(
-                zip_src, options=options, logger=self.logger
-            ).as_base64()
-        elif "namespace" in dependency:
-            self.logger.info(
-                "Installing {} version {}".format(
-                    dependency["namespace"], dependency["version"]
-                )
-            )
-            package_zip = InstallPackageZipBuilder(
-                dependency["namespace"],
-                dependency["version"],
-                securityType=self.options["security_type"],
-            )()
-
-        if package_zip:
-            api = self.api_class(
-                self, package_zip, purge_on_delete=self.options["purge_on_delete"]
-            )
-            return api()
-        elif "version_id" in dependency:
-            self.logger.info(f"Installing {dependency['version_id']}")
-            install_package_version(self.project_config, self.org_config, dependency)
-        else:
-            raise TaskOptionsError(f"Could not find package for {dependency}")
-
-    def _uninstall_dependency(self, dependency):
-        self.logger.info("Uninstalling {}".format(dependency["namespace"]))
-        package_zip = UninstallPackageZipBuilder(
-            dependency["namespace"], self.project_config.project__package__api_version
-        )
-        api = self.api_class(
-            self, package_zip(), purge_on_delete=self.options["purge_on_delete"]
-        )
-        return api()
+            dependency.install(self.project_config, self.org_config)
 
     def freeze(self, step):
         ui_options = self.task_config.config.get("ui_options", {})
-        dependencies = self.project_config.get_static_dependencies(
-            self.options["dependencies"],
-            include_beta=self.options["include_beta"],
+        dependencies = get_static_dependencies(
+            self.dependencies,
+            self.resolution_strategy,
+            self.project_config,
             ignore_deps=self.options.get("ignore_dependencies"),
         )
+
         steps = []
-        for i, dependency in enumerate(self._flatten(dependencies), start=1):
-            name = dependency.pop("name", None)
-            if "namespace" in dependency:
+        for i, dependency in enumerate(dependencies, start=1):
+            if isinstance(
+                dependency,
+                (PackageNamespaceVersionDependency, PackageVersionIdDependency),
+            ):
                 kind = "managed"
```

Comment:
We should probably change this to "package" since a version id could be different types of package -- but that will take some coordination on the metadeploy side to change the available options. This is not a blocker.

Reply:
Let's make it an enum while we're at it. I can add that WI to my list-to-create.